src/provider.ts

```diff
@@ -0,0 +1,22 @@
+import {ethers} from "ethers";
+import {EthereumNetwork} from "./types";
+
+let savedWeb3Provider = ethers.getDefaultProvider(EthereumNetwork.Default);
+let savedWallet: ethers.Wallet | undefined;
+
+export const setWeb3Provider = (provider: ethers.providers.BaseProvider) => {
```

Comment:
I am not sure about the type "BaseProvider" for a type. it doesn`t have getSigner() method on it. In TT website I am using JsonRpcProvider for the type thats why.